cli/index.ts

```diff
@@ -8,75 +8,113 @@ import * as yargs from "yargs";
 
 const RECOMPILE_DELAY = 2000;
 
-const commonBuildOptions: INamedOption[] = [
-  {
-    name: "full-refresh",
-    option: {
-      describe: "If set, this will rebuild incremental tables from scratch",
-      type: "boolean",
-      default: false,
-      alias: "fr"
-    }
-  },
-  {
-    name: "nodes",
-    option: {
-      describe: "A list of node names or patterns to run, which can include * wildcards",
-      type: "array",
-      alias: "n"
+const projectDirOption = {
+  name: "project-dir",
+  option: {
+    describe: "The Dataform project directory.",
+    default: ".",
+    coerce: dir => {
+      dir = path.resolve(dir);
+      if (!fs.existsSync(dir)) {
+        throw new Error(`Credentials file ${dir} does not exist!`);
+      }
+      return dir;
     }
-  },
-  {
-    name: "include-deps",
-    option: {
-      describe: "If set, dependencies for selected nodes will also be run",
-      type: "boolean",
-      alias: "id"
+  }
+};
+
+const fullRefreshOption = {
+  name: "full-refresh",
+  option: {
+    describe: "Forces incremental tables to be rebuilt from scratch.",
+    type: "boolean",
+    default: false
+  }
+};
+
+const nodesOption = {
+  name: "nodes",
+  option: {
+    describe: "A list of node names or patterns to run. Can include '*' wildcards.",
+    type: "array"
+  }
+};
+
+// TODO: Should this be only set when "nodes" is also set? (using yargs 'implies')
+const includeDepsOption = {
+  name: "include-deps",
+  option: {
+    describe: "If set, dependencies for selected nodes will also be run.",
+    type: "boolean"
+  }
+};
+
+const defaultSchemaOption = {
+  name: "default-schema",
+  option: {
+    describe: "An optional default schema name override."
+  }
+};
+
+const assertionSchemaOption = {
+  name: "assertion-schema",
+  option: {
+    describe: "An optional assertion schema name override."
+  }
+};
+
+const credentialsOption = {
+  name: "credentials",
+  option: {
+    describe: "The location of the credentials JSON file to use.",
+    default: ".df-credentials.json",
+    coerce: filePath => {
+      filePath = path.resolve(filePath);
+      if (!fs.existsSync(filePath)) {
+        throw new Error(`Credentials file ${filePath} does not exist!`);
+      }
+      return filePath;
```

Comment:
Maybe extract this function and use it here and in `projectDirOption`

Reply:
Done!